### PR TITLE
Add PropertyBag to TokenValidationParameters

### PR DESCRIPTION
**Pull request by GeoK**
Ease addition of custom TokenValidationParameters by utilizing
a Property Bag.

Resolves: #985
See also: #874

### CODE REVIEW

File: test/Microsoft.IdentityModel.Tokens.Tests/TokenValidationParametersTests.cs [hunk 1]
```diff
@@ -170,6 +178,7 @@ public void GetSets()
             Assert.Null(validationParameters.LifetimeValidator);
             Assert.Null(validationParameters.IssuerSigningKeyResolver);
             Assert.Null(validationParameters.IssuerValidator);
+            Assert.Null(validationParameters.PropertyBag);
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Tests/TokenValidationParametersTests.cs]: Use GetSetContext for checking default then assignment.

File: test/Microsoft.IdentityModel.Tokens.Tests/TokenValidationParametersTests.cs [hunk 2]
```diff
@@ -68,6 +68,11 @@ public void Publics()
             string validIssuer = "ValidIssuer";
             List<string> validIssuers = new List<string> { validIssuer };
 
+            IDictionary<string, Object> propertyBag =
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Tests/TokenValidationParametersTests.cs]: use var

**Comment by GeoK** [comment]
@brentschmaltz Thank you for the suggestions, I addressed them with: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/986/commits/821970eacc0f056580b384c0f6a78d2935053b0e

**Comment by brentschmaltz** [comment]
:shipit: remember to rebase and squash

**Pull request merged**